lib/on-branch-status.js

```diff
@@ -0,0 +1,229 @@
+      combined.statuses.push({
+        state: checkRun.conclusion,
+        context: checkRun.name,
+        description: checkRun.output.summary
+      })
+      return checkRun.status === 'completed' && checkRun.conclusion
+    }))
+
+    // If there are fewer conclusions than total runs, some are incomplete/pending
+    // and we can’t continue
+    if (checkRunConclusions.length !== allCheckRuns.total_count) return
+
+    // If the collected conclusions contain `cancelled`, `timed_out`, or `action_required`, we
+    // can’t really know what to do, so we bail as well. It’s probable that the check will re-run
+    // either automatically or by user action later, and at some point all conclusions should be one of
+    // `success`, `failure`, or `neutral`, which we can work with.
+    const undesirableConclusions = ['cancelled', 'timed_out', 'action_required']
+    if (_.intersection(checkRunConclusions, undesirableConclusions).length) return
+    // Finally, if one of the conclusions is `failure`, we hijack the combined state and set it to `failure` too.
+    // This means that `neutral` is NOT a failure condition. We understand it as more of an `info` type
+    // message
+    if (checkRunConclusions.includes('failure')) {
+      combined.state = 'failure'
+    }
+  }
+
+  const branchDoc = _.get(
+    await repositories.query('by_branch_sha', {
+      key: sha,
+      include_docs: true
+    }),
+    'rows[0].doc'
+  )
+  // branch was not created by Greenkeeper
+  if (!branchDoc) return
+  // branch already processed
+  if (branchDoc.processed) return
+  // state did not change
+  if (branchDoc.state === combined.state) return
+  // branch is for a node update or deprecation (we just open an issue, no PR)
+  if (branchDoc.head) {
+    const skippableBranches = ['update-to-node-', 'deprecate-node-']
+    const skipBranch = !!skippableBranches.find((skippable) => {
+      return branchDoc.head.match(RegExp(skippable, 'i'))
+    })
+    if (skipBranch) return
+  }
+
+  if (branchDoc.initial) {
+    const result = await repositories.allDocs({
+      include_docs: true,
+      descending: true,
+      startkey: `${repository.id}:pr:\uffff`,
+      endkey: `${repository.id}:pr:`
+    })
+    const initialRow = result.rows.find((row) => {
+      return row.doc.initial && row.doc.createdByUser
+    })
+
+    if (initialRow) {
+      return {
+        data: {
+          name: 'create-initial-pr-comment',
+          accountId,
+          branchDoc,
+          prDocId: initialRow.doc._id,
+          repository,
+          combined,
+          installationId: installation.id
+        }
+      }
+    }
+    return {
+      data: {
+        name: 'create-initial-pr',
+        accountId,
+        branchDoc,
+        repository,
+        combined,
+        installationId: installation.id
+      }
+    }
+  }
+
+  if (branchDoc.subgroupInitial) {
+    const result = await repositories.allDocs({
```

Comment:
I think it’s not outside the loops because you’d have to put it into an additional `if (branchDoc.initial
|| branchDoc.subgroupInitial) {}` so it doesn’t get called unnecessarily otherwise.